proposals/aggregate.json

```
@@ -0,0 +1,135 @@
+{
+    "id": "aggregate",
+    "summary": "Aggregation based on general intervals",
+    "description": "Computes a aggregation based on an array of intervals.\n\nThe computed values will be projected to the labels. If no labels are specified, the lower value of the interval will be used as label for the corresponding values. In case of a conflict (i.e. the user-specified values for the lower values of the intervals are not distinct), the user-defined labels must be specified in the parameter `labels` as otherwise a `DistinctDimensionLabelsRequired` exception would be thrown. The number of user-defined labels and the number of intervals need to be equal.",
+    "categories": [
+        "cubes",
+        "aggregate"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "data",
+            "description": "A data cube.",
+            "schema": {
+                "type": "object",
+                "subtype": "datacube"
+            }
+        },
+        {
+            "name": "intervals",
+            "description": "Left-closed intervals, which are allowed to overlap. Each interval in the array has exactly two elements:\n\n1. The first element is the lower value of the interval. The specified value is **included** in the interval.\n2. The second element is the upper value of the temporal interval. The specified value is **excluded** from the interval.\n\nThe second element must always be greater than the first element. Otherwise, an `ExtentEmpty` exception is thrown.",
```

when working with a spatial dimension (e.g. "x" or "y"): how is a users supposed to know what CRS to use to define the intervals? I don't think this is explicitly available or defined on a datacube.

---

> The second element is the upper value of the temporal interval.

I guess that "temporal" is not intentional there

---

Isn't it? The datacube extension defines the CRS and then that defines the unit and extents etc.

```suggestion
            "description": "Left-closed intervals, which are allowed to overlap. Each interval in the array has exactly two elements:\n\n1. The first element is the lower value of the interval. The specified value is **included** in the interval.\n2. The second element is the upper value of the interval. The specified value is **excluded** from the interval.\n\nThe second element must always be greater than the first element. Otherwise, an `ExtentEmpty` exception is thrown.",
```

---

> The datacube extension defines the CRS and then that defines the unit and extents etc.

Yes, but nothing that is relevant to figuring out the actual labels (`reference_system`, `step`, `values` and what not) is required (in datacube extension, nor openEO API). 

Also, this might be backend-dependend, so that would undermine the reproducibility of the process graph